src/lean_spec/subspecs/forkchoice/store.py

```diff
@@ -420,25 +485,27 @@ def produce_block(self, slot: Slot, validator_index: ValidatorIndex) -> Block:
             post_state = advanced_state.process_block(candidate_block)
 
             # Find new valid attestations matching post-state justification
-            new_attestations: list[SignedVote] = []
-            for validator_id, checkpoint in self.latest_known_votes.items():
+            new_attestations: list[ValidatorAttestation] = []
+            for signed in self.latest_known_votes.values():
                 # Skip if target block is unknown in our store
-                if checkpoint.root not in self.blocks:
+                data = signed.message.data
+                if data.target.root not in self.blocks:
```

Comment:
```suggestion
                if data.head.root not in self.blocks:
```

we need to check for `head` which is the target  of the attestation